src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -47,19 +53,33 @@ public override async Task RegisterCodeFixesAsync(CodeFixContext context)
             var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
 
             var initialNode = root.FindNode(diagnostic.Location.SourceSpan);
+            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
 
             for (var node = initialNode; node != null; node = node.Parent)
             {
-                if (node is TObjectCreationExpressionSyntax objectCreation)
-                {
-                    var argumentOpt = TryGetRelevantArgument(initialNode, node, diagnostic);
-                    await HandleObjectCreationExpressionAsync(context, objectCreation, argumentOpt).ConfigureAwait(false);
-                    return;
-                }
-                else if (node is TInvocationExpressionSyntax invocationExpression)
+                var fixData =
+                    TryGetInvocationExpressionFixInfo(semanticModel, syntaxFacts, node, cancellationToken) ??
+                    TryGetObjectCreationFixInfo(semanticModel, syntaxFacts, node, cancellationToken) ??
+                    TryGetLanguageSpecificFixInfo(semanticModel, node, cancellationToken);
+
+                if (fixData != null)
                 {
+                    var candidates = fixData.MethodCandidates;
+                    if (fixData.IsConstructorInitializer)
+                    {
+                        // The invocation is a :this() or :base() call. In  the 'this' case we need to exclude the 
+                        // method with the diagnostic because otherwise we might introduce a call to itself (which is forbidden).
+                        if (semanticModel.GetEnclosingSymbol(node.SpanStart, cancellationToken) is IMethodSymbol methodWithDiagnostic)
```

Comment:
@CyrusNajmabadi was concerned that this might not be sufficient if the call to the constructor is somehow nested. But I think it is okay because the rules are quite strict here and the enclosing symbol should always be the constructor (and even if it isn't it's not a big deal if we offer a fix that makes no sense in some odd corner cases).

Reply:
cyrus is a language expert so anything language related, he will be good person to listen to.

but that being said, as long as it won't crash VS, I am fine. if it doesn't offer a fix in a certain case or offer a wrong one, we can fix that bug when it is discovered.

Reply:
Fine. I just wanted to bring this to your attention because it was one unresolved issue buried somewhere in the thread.

Reply:
sure. thank you for doing that.